avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep

```diff
@@ -0,0 +1,1030 @@
+          {
+            secretRef: 'personal-access-token'
+            triggerParameter: 'personalAccessToken'
+          }
+          {
+            secretRef: 'organization-url'
+            triggerParameter: 'organizationURL'
+          }
+        ]
+      }
+    ]
+  : []
+
+var gitHubOrganization = selfHostedConfig.selfHostedType == 'github' ? selfHostedConfig.githubOrganization : null
+
+var gitHubRepository = selfHostedConfig.selfHostedType == 'github' ? selfHostedConfig.githubRepository : null
+
+var gitHubRunnerURL = 'https://github.com/${gitHubOrganization}/${gitHubRepository}'
+
+var devOpsOrgURL = selfHostedConfig.selfHostedType == 'azuredevops'
+  ? 'https://dev.azure.com/${selfHostedConfig.devOpsOrganization}'
+  : null
+
+// ================ //
+// Resources        //
+// ================ //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.devops-cicdagentsandrunners.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module logAnalyticsWokrspace 'br/public:avm/res/operational-insights/workspace:0.5.0' = {
+  name: 'logAnalyticsWokrspace'
+  params: {
+    name: 'law-${namingPrefix}-${uniqueString(resourceGroup().id)}-law'
+  }
+}
+
+module userAssignedIdentity 'br/public:avm/res/managed-identity/user-assigned-identity:0.3.0' = {
+  name: 'userAssignedIdentity'
+  params: {
+    name: 'msi-${namingPrefix}-${uniqueString(resourceGroup().id)}'
+    location: location
+  }
+}
+
+module acrPrivateDNSZone 'br/public:avm/res/network/private-dns-zone:0.5.0' = if (privateNetworking && empty(networkingConfiguration.?containerRegistryPrivateDnsZoneId ?? '')) {
+  name: 'acrdnszone${namingPrefix}${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'privatelink.azurecr.io'
+    virtualNetworkLinks: [
+      {
+        virtualNetworkResourceId: networkingConfiguration.networkType == 'createNew'
+          ? newVnet.outputs.resourceId
+          : networkingConfiguration.virtualNetworkResourceId
+      }
+    ]
+  }
+}
+
+module acr 'br/public:avm/res/container-registry/registry:0.4.0' = {
+  name: 'acr${namingPrefix}${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'acr${namingPrefix}${uniqueString(resourceGroup().id)}'
+    acrSku: privateNetworking ? 'Premium' : 'Standard'
+    acrAdminUserEnabled: false
+    roleAssignments: [
+      {
+        principalId: userAssignedIdentity.outputs.principalId
+        roleDefinitionIdOrName: '7f951dda-4ed3-4680-a7ca-43fe172d538d'
```

Comment:
Would be create to add a comment to all role assignment that specifies the role you're assigning. GUIDs are hard to interpret 😉

Reply:
Resolved